### PR TITLE
fix: cleanup the KMS validation logic

### PR DESCRIPTION
**Pull request by ocofaigh**
### Description

Multiple updates - see release notes section below

NOTE: I had to skip the upgrade test because I changed the default values used for resource allocation. I manually verified the complete example ran locally (this is the example used for upgrade test)

#### Changes that affect the core Terraform module or submodules
- [x] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [x] Examples or tests (addition or updates of examples or tests)
- [x] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

##### Release notes content

- `cpu_count` default value is now 0, meaning shared CPU by default. When updating from a previous version, you may now wish to explicitly set this to match (or update the current value). Previous default was 7.
- `disk_mb` default value now `10240` (was previously `20480`).
- `kms_encryption_enabled` must now be set if you want to use BYOK or KYOK encryption
- `skip_iam_authorization_policy` now defaults to `true`, but is only relevant if `kms_encryption_enabled` is set to a`true`
- The input variable `allowlist` has been removed as its now a legacy. The recommendation is to use CBRs instead.
- variable validation and description cleanup
- added support to create service credentials using new variable 

FSCloud profile specific changes:
- `backup_encryption_key_crn` is no longer a supported variable since it doesn't support HPCS (profile defaults to ICD randomly generated encryption key)
- added new outputs: `crn`, `cbr_rule_ids`, `service_credentials_json` and `service_credentials_object`
- default value for `plan` is now enterprise (was previously standard)
- Due to enterprise now being the default, the following variable defaults have changed:
    - `memory_mb`: 1024 -> 14336
- auto_scaling defaults to `null` (meaning disabled by default)

**BREAKING CHANGES**

* If you were using the legacy input variable `allowlist` to restrict network to a specific IP range, you will need to migrate to using Context Based Restrictions (CBRs) for this. For more info see [Protecting Cloud Databases resources with context-based restrictions](https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-cbr&interface=ui)

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: profiles/fscloud/README.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
+The default values in this profile were scanned by [IBM Code Risk Analyzer (CRA)](https://cloud.ibm.com/docs/code-risk-analyzer-cli-plugin?topic=code-risk-analyzer-cli-plugin-cra-cli-plugin#terraform-command) for compliance with the IBM Cloud Framework for Financial Services profile that is specified by the IBM Security and Compliance Center. The scan passed for all applicable goals with one exception:
+
+> rule-beb7b289-706b-4dc0-b01d-b1d15d4331e3: Check whether Databases for MongoDB network access is restricted to a specific IP range
+
+The IBM Cloud Framework for Financial Services mandates the application of an inbound network-based allowlist in front of the ICD MongoDB instance. You can comply with this requirement by using the `cbr_rules` variable in the module, which can be used to create a narrow context-based restriction rule that is scoped to the IBM Cloud Databases for MongoDB instance. CRA does not currently support checking for context-based restrictions, so you can ignore the failing CRA rule as long as you have set the context-based restriction.
```
Comment (SirSpidey) [inline profiles/fscloud/README.md:9]:
```suggestion
The IBM Cloud Framework for Financial Services mandates the application of an inbound network-based allowlist in front of the ICD MongoDB instance. You can comply with this requirement by using the `cbr_rules` variable in the module, which can be used to create a narrow context-based restriction rule that is scoped to the IBM Cloud Databases for MongoDB instance. CRA does not currently support checking for context-based restrictions, so you can ignore the failing rule after you set the context-based restriction.
```

Syncing this with the postgresql model

File: profiles/fscloud/variables.tf [hunk 2]
```diff
@@ -3,30 +3,24 @@
...
 
-variable "plan" {
-  type        = string
-  description = "The name of the service plan that you choose for your mongodb instance"
-  default     = "standard"
-}
-
 variable "region" {
-  description = "The region mongodb is to be created on. The region must support KYOK."
+  description = "The region where you want to deploy your instance. Must be the same region as the Hyper Protect Crypto Service."
```
Comment (SirSpidey) [inline profiles/fscloud/variables.tf:17]:
```suggestion
  description = "The region where you want to deploy your instance. Must be the same region as the Hyper Protect Crypto Services."
```

File: profiles/fscloud/variables.tf [hunk 3]
```diff
@@ -37,24 +31,57 @@ variable "configuration" {
...
+    ])
+    error_message = "Only supported plans are standard or enterprise"
+  }
 }
 
 variable "memory_mb" {
-  description = "Memory available to the mongodb instance"
   type        = number
-  default     = 1024
+  description = "Allocated memory per-member. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```
Comment (SirSpidey) [inline profiles/fscloud/variables.tf:49]:
```suggestion
  description = "Allocated memory per member. For more information, see https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```

File: profiles/fscloud/variables.tf [hunk 4]
```diff
@@ -37,24 +31,57 @@ variable "configuration" {
...
+  default     = 14336
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
 }
 
 variable "disk_mb" {
-  description = "Disk space available to the mongodb instance"
   type        = number
+  description = "Allocated disk per-member. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```
Comment (SirSpidey) [inline profiles/fscloud/variables.tf:56]:
```suggestion
  description = "Allocated disk per member. For more information, see https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```

File: profiles/fscloud/README.md (hunk 1 above)
Comment (ocofaigh) [reply profiles/fscloud/README.md:9]: thanks, you beat me to it :)

File: profiles/fscloud/variables.tf [hunk 5]
```diff
@@ -63,66 +90,57 @@ variable "instance_name" {
...
-variable "backup_encryption_key_crn" {
+  description = "The GUID of the Hyper Protect Crypto Service."
   type        = string
-  description = "The CRN of a Key Protect Key to use for encrypting backups. Take note that Hyper Protect Crypto Services for IBM Cloud® Databases backups is not currently supported."
 }
 
 variable "skip_iam_authorization_policy" {
   type        = bool
-  description = "Set to true to skip the creation of an IAM authorization policy that permits all mongodb instances in the provided resource group reader access to the instance specified in the existing_kms_instance_guid variable."
+  description = "Set to true to skip the creation of an IAM authorization policy that permits all MongoDB database instances in the given resource group to read the encryption key from the Hyper Protect instance passed in var.existing_kms_instance_guid."
```
Comment (SirSpidey) [inline profiles/fscloud/variables.tf:109]:
```suggestion
  description = "Set to true to skip the creation of an IAM authorization policy that permits all MongoDB database instances in the resource group to read the encryption key from the Hyper Protect Crypto Services instance. The HPCS instance is passed in through the var.existing_kms_instance_guid variable."
```
Trying to simplify the definition by shortening the sentence.

File: variables.tf [hunk 6]
```diff
@@ -168,12 +148,60 @@ variable "auto_scaling" {
...
+
+variable "kms_encryption_enabled" {
+  type        = bool
+  description = "Set this to true to control the encryption keys used to encrypt the data that you store in IBM Cloud® Databases. If set to false, the data is encrypted by using randomly generated keys. For more info on Key Protect integration, see https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-key-protect. For more info on HPCS integration, see https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-hpcs"
+  default     = false
+}
+
+variable "kms_key_crn" {
+  type        = string
+  description = "The root key CRN of a Key Management Services like Key Protect or Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption. Only used if var.kms_encryption_enabled is set to true."
```
Comment (SirSpidey) [inline variables.tf:167]:
```suggestion
  description = "The root key CRN of a Key Management Services like Key Protect or Hyper Protect Crypto Services (HPCS) that you want to use for disk encryption. Only used if var.kms_encryption_enabled is set to true."
```

File: variables.tf [hunk 7]
```diff
@@ -168,12 +148,60 @@ variable "auto_scaling" {
...
+      can(regex(".*kms.*", var.kms_key_crn)),
+      can(regex(".*hs-crypto.*", var.kms_key_crn)),
+    ])
+    error_message = "Value must be the root key CRN from either the Key Protect or Hyper Protect Crypto Service (HPCS)"
+  }
+}
+
+variable "backup_encryption_key_crn" {
+  type        = string
+  description = "The CRN of a Key Protect key, that you want to use for encrypting disk that holds deployment backups. Only used if var.kms_encryption_enabled is set to true. If no value passed, the value passed for the 'kms_key_crn' variable will be used. BYOK for backups is available only in US regions us-south and us-east, and eu-de. Only keys in the us-south and eu-de are durable to region failures. To ensure that your backups are available even if a region failure occurs, you must use a key from us-south or eu-de. Take note that Hyper Protect Crypto Services for IBM Cloud® Databases backups is not currently supported, so if no value is passed here, but a HPCS value is passed for var.kms_key_crn, databases backup encryption will use the default encryption keys."
```
Comment (SirSpidey) [inline variables.tf:181]:
```suggestion
  description = "The CRN of a Key Protect key that you want to use for encrypting the disk that holds deployment backups. Only used if var.kms_encryption_enabled is set to true. BYOK for backups is available only in US regions us-south and us-east, and in eu-de. Only keys in the us-south and eu-de are durable to region failures. To ensure that your backups are available even if a region failure occurs, use a key from us-south or eu-de. Hyper Protect Crypto Services for IBM Cloud Databases backups is not currently supported. If no value is passed here, the value passed for the 'kms_key_crn' variable is used. And if a HPCS value is passed for var.kms_key_crn, the database backup encryption uses the default encryption keys."
```

Trying to make sense of this complex definition.

File: variables.tf [hunk 8]
```diff
@@ -168,12 +148,60 @@ variable "auto_scaling" {
...
+}
+
+variable "skip_iam_authorization_policy" {
+  type        = bool
+  description = "Set to true to skip the creation of an IAM authorization policy that permits all MongoDB database instances in the given resource group to read the encryption key from the Hyper Protect or Key Protect instance passed in var.existing_kms_instance_guid. If set to 'false', a value must be passed for var.existing_kms_instance_guid. No policy is created if var.kms_encryption_enabled is set to 'false'."
+  default     = false
+}
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Hyper Protect or Key Protect instance in which the key specified in var.kms_key_crn and var.backup_encryption_key_crn is coming from. Only required if var.kms_encryption_enabled is 'true', var.skip_iam_authorization_policy is 'false', and passing a value for var.kms_key_crn and/or var.backup_encryption_key_crn."
```
Comment (SirSpidey) [inline variables.tf:196]:
```suggestion
  description = "The GUID of the Hyper Protect Crypto Services or Key Protect instance in which the key specified in var.kms_key_crn and var.backup_encryption_key_crn is coming from. Required only if var.kms_encryption_enabled is set to true, var.skip_iam_authorization_policy is set to false, and you pass a value for var.kms_key_crn, var.backup_encryption_key_crn, or both."
```

File: variables.tf [hunk 9]
```diff
@@ -168,12 +148,60 @@ variable "auto_scaling" {
...
+  default     = null
+  validation {
+    condition     = var.backup_encryption_key_crn == null ? true : length(regexall("^crn:v1:bluemix:public:kms:(us-south|us-east|eu-de):a/[[:xdigit:]]{32}:[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}:key:[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}$", var.backup_encryption_key_crn)) > 0
+    error_message = "Valid values for backup_encryption_key_crn is null or a Key Protect key CRN from us-south, us-east or eu-de"
+  }
+}
+
+variable "skip_iam_authorization_policy" {
+  type        = bool
+  description = "Set to true to skip the creation of an IAM authorization policy that permits all MongoDB database instances in the given resource group to read the encryption key from the Hyper Protect or Key Protect instance passed in var.existing_kms_instance_guid. If set to 'false', a value must be passed for var.existing_kms_instance_guid. No policy is created if var.kms_encryption_enabled is set to 'false'."
```
Comment (SirSpidey) [inline variables.tf:191]:
```suggestion
  description = "Set to true to skip the creation of an IAM authorization policy that permits all MongoDB database instances in the resource group to read the encryption key from the Hyper Protect Crypto Services (HPCS) instance. If set to false, pass in a value for the HPCS instance in the  var.existing_kms_instance_guid variable. In addition, no policy is created if var.kms_encryption_enabled is set to false."
```
phew. complex. I removed some stuff that I think is redundant

**Review by SirSpidey — COMMENTED** [review]
I'm probably spending too much time on this, but figured I'd review some descriptions here that might be used in other ICD modules.

File: profiles/fscloud/variables.tf [hunk 10]
```diff
@@ -63,66 +90,57 @@ variable "instance_name" {
...
+  description = "The root key CRN of the Hyper Protect Crypto Service (HPCS) to use for disk encryption."
 }
 
 variable "existing_kms_instance_guid" {
-  description = "The GUID of the Hyper Protect Crypto service."
-  type        = string
-}
-
-variable "backup_encryption_key_crn" {
+  description = "The GUID of the Hyper Protect Crypto Service."
```
[the remark this answers is not in the document]
Comment (ocofaigh) [reply profiles/fscloud/variables.tf:103]: Key Protect is not FSCloud compliant, so its not supported in the fscloud module. I'll update to "The GUID of the Hyper Protect Crypto Services instance."

**Comment by ocofaigh** [comment]
Thanks @SirSpidey that is just what I needed so I can roll out consistency to all other ICD modules

File: examples/fscloud/README.md [hunk 11]
```diff
@@ -1,17 +1,18 @@
...
-- Create an IAM Authorization between MongoDB instance resource group and Key Protect Instance for backup_encryption_key_crn as backup encryption key is not supported by Hyper Protect instances yet.
-- Create an IAM Authorization between MongoDB instance Resource group and HPSC permanent Instance.
-- Create a new ICD MongoDB instance and credentials that is encrypted using the Hyper Protect Crypto Service resources that are passed in.
-- Create a Sample VPC.
-- Create Context Based Restriction(CBR) to only allow MongoDB to be accessible from the VPC.
+:exclamation: **Important:** In this example, only the IBM Cloud Databases for MongoDB instance complies with the IBM Cloud Framework for Financial Services. Other parts of the infrastructure do not necessarily comply.
+
+## Before you begin
+
+- You need a Hyper Protect Crypto Service instance and root key available in the region that you want to deploy your MongoDB database instance to.
```
Comment (SirSpidey) [inline examples/fscloud/README.md:18]:
```suggestion
- You need a Hyper Protect Crypto Services instance and root key available in the region that you want to deploy your MongoDB database instance to.
```

File: examples/fscloud/variables.tf [hunk 12]
```diff
@@ -40,6 +40,17 @@ variable "kms_key_crn" {
 
 variable "mongodb_version" {
   type        = string
-  description = "Version of the mongodb instance. If left at null, the latest version is provisioned."
+  description = "Version of the mongodb instance. If no value passed, the current ICD preferred version is used."
```
Comment (SirSpidey) [inline examples/fscloud/variables.tf:43]:
```suggestion
  description = "Version of the MongoDB instance. If no value is passed, the current preferred version of IBM Cloud Databases is used."
```

File: profiles/fscloud/README.md [hunk 13]
```diff
@@ -1,18 +1,20 @@
...
+The default values in this profile were scanned by [IBM Code Risk Analyzer (CRA)](https://cloud.ibm.com/docs/code-risk-analyzer-cli-plugin?topic=code-risk-analyzer-cli-plugin-cra-cli-plugin#terraform-command) for compliance with the IBM Cloud Framework for Financial Services profile that is specified by the IBM Security and Compliance Center. The scan passed for all applicable goals with one exception:
+
+> rule-beb7b289-706b-4dc0-b01d-b1d15d4331e3: Check whether Databases for MongoDB network access is restricted to a specific IP range
+
+The IBM Cloud Framework for Financial Services mandates the application of an inbound network-based allowlist in front of the ICD MongoDB instance. You can comply with this requirement by using the `cbr_rules` variable in the module, which can be used to create a narrow context-based restriction rule that is scoped to the IBM Cloud Databases for MongoDB instance. CRA does not currently support checking for context-based restrictions, so you can ignore the failing rule after you set the context-based restriction.
```
Comment (SirSpidey) [inline profiles/fscloud/README.md:9]:
```suggestion
The IBM Cloud Framework for Financial Services mandates the application of an inbound network-based allowlist in front of the IBM Cloud Databases for (ICD) MongoDB instance. You can comply with this requirement by using the `cbr_rules` variable in the module, which can be used to create a narrow context-based restriction rule that is scoped to the MongoDB instance. CRA does not currently support checking for context-based restrictions, so you can ignore the failing rule after you set the context-based restriction.
```

File: profiles/fscloud/main.tf [hunk 14]
```diff
@@ -6,13 +6,15 @@ module "mongodb" {
   skip_iam_authorization_policy = var.skip_iam_authorization_policy
   endpoints                     = "private"
   mongodb_version               = var.mongodb_version
+  kms_encryption_enabled        = true
   existing_kms_instance_guid    = var.existing_kms_instance_guid
   kms_key_crn                   = var.kms_key_crn
-  backup_encryption_key_crn     = var.backup_encryption_key_crn
+  backup_encryption_key_crn     = null # Need to use default encryption until ICD HPCS adds support for backup encryption
```
Comment (SirSpidey) [inline profiles/fscloud/main.tf:12]: Not critical, but I'm not aware of a service called "ICD HPCS".

File: examples/complete/variables.tf [hunk 15]
```diff
@@ -30,20 +30,19 @@ variable "resource_tags" {
 
 variable "mongodb_version" {
   type        = string
-  description = "Version of the mongodb instance. If left at null, the latest version is provisioned."
+  description = "Version of the mongodb instance. If no value passed, the current ICD preferred version is used."
```
Comment (SirSpidey) [inline examples/complete/variables.tf:33]:
```suggestion
  description = "Version of the MongoDB instance. If no value is passed, the current preferred version of IBM Cloud Databases is used."
```

File: profiles/fscloud/variables.tf [hunk 16]
```diff
@@ -3,30 +3,24 @@
...
 
 variable "resource_group_id" {
-  description = "The resource group ID where the mongodb will be created"
   type        = string
+  description = "The resource group ID where the MongoDB instance will be created."
 }
 
 variable "mongodb_version" {
-  description = "The version of mongodb. If null, the current default ICD mongodb version is used."
+  description = "The version of the MongoDB to provision. If no value passed, the current ICD preferred version is used."
```
Comment (SirSpidey) [inline profiles/fscloud/variables.tf:11]:
```suggestion
  description = "Version of the MongoDB instance. If no value is passed, the current preferred version of IBM Cloud Databases is used."
```

File: variables.tf [hunk 17]
```diff
@@ -54,32 +46,32 @@ variable "configuration" {
...
+  description = "The version of the MongoDB to provision. If no value passed, the current ICD preferred version is used."
   default     = null
   validation {
     condition = anytrue([
       var.mongodb_version == null,
       var.mongodb_version == "5.0",
       var.mongodb_version == "4.4",
     ])
-    error_message = "Version must be 4.4 or 5.0. If null, the current default ICD mongodb version is used"
+    error_message = "Version must be 4.4 or 5.0. If no value passed, the current ICD preferred version is used."
```
Comment (SirSpidey) [inline variables.tf:57]:
```suggestion
    error_message = "Version must be 4.4 or 5.0. If no value is passed, the current preferred version of IBM Cloud Databases is used."
```

File: variables.tf [hunk 18]
```diff
@@ -88,54 +80,42 @@ variable "endpoints" {
...
+  description = "Allocated disk per-member. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
+  default     = 10240
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
 }
 
 variable "cpu_count" {
   type        = number
-  description = "The number of CPU cores available to the database instance. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
-  default     = 7
+  description = "Allocated dedicated CPU per-member. For shared CPU, set to 0. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```
Comment (SirSpidey) [inline variables.tf:99]:
```suggestion
  description = "Allocated dedicated CPU per member. For shared CPU, set to 0. For more information, see https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```

File: variables.tf [hunk 19]
```diff
@@ -88,54 +80,42 @@ variable "endpoints" {
...
-      can(regex(".*kms.*", var.backup_encryption_key_crn))
-    ])
-    error_message = "A Key Protect key is required for encryption of backups"
-  }
-}
-
 variable "memory_mb" {
   type        = number
-  description = "Memory available to the database instance in MB. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
+  description = "Allocated memory per-member. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```
Comment (SirSpidey) [inline variables.tf:85]:
```suggestion
  description = "Allocated memory per member. For more information, see https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```

File: variables.tf [hunk 20]
```diff
@@ -88,54 +80,42 @@ variable "endpoints" {
...
   default     = 1024
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
 }
 
 variable "disk_mb" {
   type        = number
-  description = "Disk space available to the database instance. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
-  default     = 20480
+  description = "Allocated disk per-member. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```
Comment (SirSpidey) [inline variables.tf:92]:
```suggestion
  description = "Allocated disk per member. For more information, see https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
```

File: variables.tf [hunk 21]
```diff
@@ -88,54 +80,42 @@ variable "endpoints" {
...
-    error_message = "A Key Protect key is required for encryption of backups"
-  }
-}
-
 variable "memory_mb" {
   type        = number
-  description = "Memory available to the database instance in MB. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
+  description = "Allocated memory per-member. For more information refer to the docs https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member"
   default     = 1024
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
```
Comment (SirSpidey) [inline variables.tf:87]:
```suggestion
  # Validation is done in the Terraform plan phase by the IBM provider, so no need to add extra validation here.
```

File: variables.tf [hunk 22]
```diff
@@ -88,54 +80,42 @@ variable "endpoints" {
...
+  default     = 0
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
 }
 
-
 variable "members" {
   type        = number
   description = "Allocated number of members"
   default     = 3
+  # Validation is done in terraform plan phase by IBM provider, so no need to add any extra validation here
```
Comment (SirSpidey) [inline variables.tf:108]:
```suggestion
  # Validation is done in the Terraform plan phase by the IBM provider, so no need to add extra validation here.
```

**Review by SirSpidey — COMMENTED** [review]
More suggestions for consistency.

**Review by toddgiguere — APPROVED** [review]

**Pull request merged**